src/backend/distributed/planner/merge_planner.c

```diff
@@ -824,6 +830,20 @@ ConvertRelationRTEIntoSubquery(Query *mergeQuery, RangeTblEntry *sourceRte,
 	RangeTblEntry *newRangeTableEntry = copyObject(sourceRte);
 	sourceResultsQuery->rtable = list_make1(newRangeTableEntry);
 
+#if PG_VERSION_NUM >= PG_VERSION_16
+	sourceResultsQuery->rteperminfos = NIL;
+	if (newRangeTableEntry->perminfoindex)
+	{
+		/* create permission info for newRangeTableEntry */
+		RTEPermissionInfo *perminfo = getRTEPermissionInfo(mergeQuery->rteperminfos,
+														   newRangeTableEntry);
+
+		/* update the subquery's rteperminfos accordingly */
+		newRangeTableEntry->perminfoindex = 1;
```

Comment:
The comment says `/* update the subquery's rteperminfos accordingly */`. Wouldn't subqueries always have = 0? I think this is a releation

Reply:
I meant to write sourceResultsQuery instead of subquery, thanks.